pkg/kubelet/pod_workers.go

```diff
@@ -1161,9 +1164,13 @@ func (p *podWorkers) removeTerminatedWorker(uid types.UID) {
 		return
 	}
 
-	if startedUID, started := p.startedStaticPodsByFullname[status.fullname]; started && startedUID != uid {
-		klog.V(4).InfoS("Pod cannot start yet but is no longer known to the kubelet, finish it", "podUID", uid)
-		status.finished = true
+	_, waiting := p.waitingToStartStaticPodsByFullname[status.fullname]
```

Comment:
You can't add things to this map unless they are static pods.  So this needs to be inside an "if static pod" block.

Reply:
Will clarify and add comments.